### ISSUE TITLE
feature: pass axios config in the storage

### ISSUE THREAD

**Issue by Warxcell** [arthurfiorette/axios-cache-interceptor#220]
Use case is to add tags to cache depending on the request.

**Comment by arthurfiorette** [comment]
Adding a method like a onConfigReady that accepts the resolved config as parameter is sufficient?

**Comment by Warxcell** [comment]
But wouldn't that make usage of buildStorage a bit harder, because itsn't not class and doesn't have this scope?

**Comment by arthurfiorette** [comment]
Yeah, probably. buildStorage parameter can be a function that returns the storage object and receives the config obj.

You're up to create a PR?

**Comment by Warxcell** [comment]
That sounds good I think. Will create a PoC.

**Comment by Warxcell** [comment]
Can you verify that I understood you correctly? https://github.com/arthurfiorette/axios-cache-interceptor/pull/221/files

**Comment by arthurfiorette** [comment]
Well, I was is a lot of meetings this morning, so I couldn't clarify you better. Sorry about that, but, I thought it would be a more clean approach to code it in this way:

```ts
buildStorage((conf) => ({
  find: ...,
  set: ...,
  remove: ...,
}))
```

And build storage also returns a function, and the type of `CacheOptions.storage` be overritten to `(CacheOptions) => AxiosStorage` (circulally referencing itself).

And, in this line, it should be passed the whole options as parameter.

https://github.com/arthurfiorette/axios-cache-interceptor/blob/d75fef44dbe97c820f6bde9480518ba9b05793c8/src/cache/create.ts#L55

**Comment by arthurfiorette** [comment]
Wait, you meant the global axios config (`AxiosInstance` + `AxiosRequestConfig` defaults) or the per-request config (and/or) response.

**Comment by Warxcell** [comment]
Per request config.

**Comment by arthurfiorette** [comment]
Can't you do that with `cache.update`?

**Comment by Warxcell** [comment]
But it requires specific IDs? How can I use it for all keys?

**Comment by Warxcell** [comment]
And actually it won't work, since I receive cached object - i have to work-around it by adding temporary tags into cache, then removing them in the storage, before storing value.

**Comment by arthurfiorette** [comment]
That won't work because the request interceptor also uses the `.set` method...

But, probably, everything that you need lives under `response.config` or `CacheRequestConfig`.

Take a look at [`220-example`](https://github.com/arthurfiorette/axios-cache-interceptor/tree/220-example). The tests aren't working but the source seems to be.

https://github.com/arthurfiorette/axios-cache-interceptor/blob/cbcccfae972361b82558a673d762b9efd0cd1cd8/src/storage/types.ts#L84-L95

**Comment by Warxcell** [comment]
Yes, that would also work aswell.

**Comment by arthurfiorette** [comment]
Hey @Warxcell, sorry for this delayed fix. I created #226 that implements what I said above. Can you check it?

As soon as you confirm, I'll release your feature.

**Comment by Warxcell** [comment]
Hey, no need to sorry :) Looks great !

**Comment by arthurfiorette** [comment]
Hey @Warxcell, I've just released 0.10.3. Can you try it?

**Comment by Warxcell** [comment]
Just tried, works as charm ! Thank you for your fast response !
